### DISCUSSION TITLE
Linking to respective github repository from dependency graph

### DISCUSSION THREAD

**Discussion by jojomatik** [community/community#26099] [Packages]
<p>Hi, I’m currently experimenting with using maven to manage my dependencies.</p>
<p>Installing the project works fine but I’m somewhat concerned whether or not my dependency graph shows the correct information.</p>
<p>I followed this guide:<br>
“Configuring Apache Maven for use with GitHub Packages” (I can only add two links to this post; I’m sorry :wink: )</p>
<p>Most dependencies such as gson, jetty etc. I currently use are clickable in the dependencies section of the dependency tree and lead to the respective github repo. My own dependencies for which I have setup their own repositories and added packages for however only show up as: “groupId:artifactId” and are not clickable. I tried this with private as well as public dependcies. (But my dependents are private at least for now)</p>
<p>So far I’ve experimented with adding a  tag  and adding a  tag to the dependencie’s pom.xml.</p>
<p>These two questions seem kind of related but due to a lack of information (or understanding on my side) I’m not sure whether or not the packages are hosted on github (doesn’t seem like they are):</p>
<p><a href="https://github.community/t/how-to-make-repository-linked-in-the-dependency-graph/3039">https://github.community/t/how-to-make-repository-linked-in-the-dependency-graph/3039</a></p>
<p><a href="https://github.community/t/how-to-get-a-project-to-show-up-in-dependency-graph/1530">https://github.community/t/how-to-get-a-project-to-show-up-in-dependency-graph/1530</a></p>
<p>I do not want to offend anyone but at least for me both of the answers don’t seem very helpful (and at least in the second post, it’s just a guess). But that might just be me not having enough understanding of the inner workins of maven and/ or package distribution.</p>
<p>The repo I want to use as a dependency is currently public, however this is just the case for testing purposes. Therefore I’d appreciate if one could provide a general answer independent from my repo as its going to be deleted.</p>
<p>I’d like to know whether linking is even possible and which parts of both pom.xml (dependent &amp; dependency) I’d have to add to do this.</p>
<p>Thank you in advance. If you need additional information, feel free to ask for further information.</p>

**Comment by jojomatik** [comment]
<p>I’m not sure if I’m supposed to add information to my post by replying, but I can’t find an option to edit the original:</p>
<p>I’ve just found <a href="https://github.community/t/download-from-github-package-registry-without-authentication/14407">this post</a> and I’m thinking that it’s probably related. Before I didn’t even notice that downloading a package via Maven required authentication as my primary PC already has authentication setup for publishing.</p>
<p>This leads me to the conclusion that github is probably not able to read the meta-information in the pom.xml of the dependency and therefore can’t read the &lt;url&gt; tag.</p>
<p>Feel free to prove me wrong but I think that mentioned issue has to be resolved before the link can show up.</p>

**Comment by whitneyimura** [comment, marked as answer]
<p>Hi <a class="mention" href="/jojomatik">@jojomatik</a> that’s correct.  Without anonymous (unauthenticated public package pulling), the link will not show up.</p>

**Reply by guylevy** [reply]
It seems to be the case also when the private packages are hosted on a GitHub packages and even if its on the same organization which is surprising and disappointing - I would expect that GitHub can fetch meta data in the dependency graph for private packages hosted in GitHub within the same organization
